### PR TITLE
Document 0, 90, 180, and 270 degree TileMap cell rotations

### PR DESCRIPTION
**Pull request by Goldenlion5648**
documents 0, 90, 180, and 270 degree TileMap cell rotations

### CODE REVIEW

File: doc/classes/TileSetAtlasSource.xml [hunk 1]
```diff
@@ -295,6 +295,16 @@
...
+			[b]Note:[/b] These transformations can be combined to do the equivalent of 0, 90, 180, and 270 degree rotations, as shown below
+			[codeblock]
+			const rotation_settings = [
+			    0,                                                                             #rotate   0
+			    TileSetAtlasSource.TRANSFORM_TRANSPOSE |  TileSetAtlasSource.TRANSFORM_FLIP_H, #rotate  90
+			    TileSetAtlasSource.TRANSFORM_FLIP_H    |  TileSetAtlasSource.TRANSFORM_FLIP_V, #rotate 180 
+			    TileSetAtlasSource.TRANSFORM_TRANSPOSE |  TileSetAtlasSource.TRANSFORM_FLIP_V  #rotate 270
+			]
+			[/codeblock]
+
```
Comment (Calinou) [inline doc/classes/TileSetAtlasSource.xml:307]:
```suggestion
			[b]Note:[/b] These transformations can be combined to do the equivalent of 0, 90, 180, and 270 degree rotations, as shown below:
			[codeblock]
			const rotation_settings = [
			    0,                                                                             # Don't rotate.
			    TileSetAtlasSource.TRANSFORM_TRANSPOSE | TileSetAtlasSource.TRANSFORM_FLIP_H,  # Rotate 90 degrees.
			    TileSetAtlasSource.TRANSFORM_FLIP_H | TileSetAtlasSource.TRANSFORM_FLIP_V,  # Rotate 180 degrees.
			    TileSetAtlasSource.TRANSFORM_TRANSPOSE | TileSetAtlasSource.TRANSFORM_FLIP_V,  # Rotate 270 degrees.
			]
			[/codeblock]
```

**Comment by mhilbrunner** [comment]
Thanks for contributing :) The commits need to be squashed into one before this can be merged. See [here](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase) for one explanation on how to do do that.

File: doc/classes/TileSetAtlasSource.xml [hunk 2]
```diff
@@ -295,6 +295,15 @@
...
 			    $TileMap.set_cell(0, Vector2i(2, 2), source_id, atlas_coords, alternate_id | TileSetAtlasSource.TRANSFORM_FLIP_H)
 			[/codeblock]
+			[b]Note:[/b] These transformations can be combined to do the equivalent of 0, 90, 180, and 270 degree rotations, as shown below:
+			[codeblock]
+			const rotation_settings = [
+			    0,  # Don't rotate.
+			    TileSetAtlasSource.TRANSFORM_TRANSPOSE | TileSetAtlasSource.TRANSFORM_FLIP_H,  # Rotate 90 degrees.
+			    TileSetAtlasSource.TRANSFORM_FLIP_H | TileSetAtlasSource.TRANSFORM_FLIP_V,  # Rotate 180 degrees.
+			    TileSetAtlasSource.TRANSFORM_TRANSPOSE | TileSetAtlasSource.TRANSFORM_FLIP_V,  # Rotate 270 degrees.
+			]
```
Comment (dalexeev) [inline doc/classes/TileSetAtlasSource.xml:305]:
```suggestion
			enum TileTransform {
			    ROTATE_0 = 0,
			    ROTATE_90 = TileSetAtlasSource.TRANSFORM_TRANSPOSE | TileSetAtlasSource.TRANSFORM_FLIP_H,
			    ROTATE_180 = TileSetAtlasSource.TRANSFORM_FLIP_H | TileSetAtlasSource.TRANSFORM_FLIP_V,
			    ROTATE_270 = TileSetAtlasSource.TRANSFORM_TRANSPOSE | TileSetAtlasSource.TRANSFORM_FLIP_V,
			}
```

**Comment by Goldenlion5648** [comment]
Is the job failure due to something on my end? Or is this PR good to go?

**Review by mhilbrunner — APPROVED** [review]
Looks good to go to me.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by Goldenlion5648** [comment]
Thanks! Will this show up in the current webpage version of the docs when the next minor release for 4.2 comes out? https://docs.godotengine.org/en/4.2/classes/class_tilesetatlassource.html

**Comment by akien-mga** [comment]
Yes, after it gets cherry-picked it will be included in the online docs too.

**Pull request merged**